napari/experimental/_progressive_loading.py

```diff
@@ -0,0 +1,1649 @@
+
+    # TODO global worker usage is not viable for real implementation
+    # Terminate existing multiscale render pass
+    if worker:
+        # TODO this might not terminate threads properly
+        worker.await_workers()
+        # worker.await_workers(msecs=30000)
+
+    # Find the corners of visible data in the highest resolution
+    corner_pixels = root_layer.corner_pixels
+
+    top_left = np.max((corner_pixels,), axis=0)[0, :]
+    bottom_right = np.min((corner_pixels,), axis=0)[1, :]
+
+    camera = viewer.camera.copy()
+
+    # TODO Added to skip situations when 3D isnt setup on layer yet??
+    if np.any((bottom_right - top_left) == 0):
+        return
+
+    # TODO we could add padding around top_left and bottom_right to account
+    #      for future camera movement
+
+    # Interval must be nonnegative
+    if not np.all(top_left <= bottom_right):
+        import pdb
+
+        pdb.set_trace()
+
+    LOGGER.info(
+        f"dims_update_handler: start render_sequence {corner_pixels} on {root_layer}"
+    )
+
+    # Find the visible scales
+    visible_scales = [False] * len(data.arrays)
+    min_scale = 0
+    max_scale = len(data.arrays) - 1
+
+    ndisplay = ndisplay if ndisplay else viewer.dims.ndisplay
+
+    # Get the scale visibility predicate for the correct ndisplay
+    should_render_scale = (
+        should_render_scale_2D if ndisplay == 2 else should_render_scale_3D
+    )
+
+    for scale in range(len(data.arrays)):
+        layer_name = get_layer_name_for_scale(scale)
+        layer = viewer.layers[layer_name]
+
+        layer.metadata["translated"] = False
+
+        # Reenable visibility of layer
+        visible_scales[scale] = should_render_scale(
+            scale, viewer, min_scale, max_scale
+        )
+        layer.visible = visible_scales[scale]
+        layer.opacity = 0.9
+
+        LOGGER.info(
+            f"scale {scale} name {layer_name}\twith translate \
+            {layer.data.translate}"
+        )
+
+    # Update the MultiScaleVirtualData memory backing
+    data.set_interval(top_left, bottom_right, visible_scales=visible_scales)
+
+    # Start a new multiscale render
+    worker = render_sequence(
+        corner_pixels,
+        data=data,
+        visible_scales=visible_scales,
+        ndisplay=ndisplay,
+        camera=camera,
+    )
+
+    LOGGER.info(
+        f"dims_update_handler: started render_sequence with corners {corner_pixels}"
+    )
+
+    # This will consume our chunks and update the numpy "canvas" and refresh
+    def on_yield(coord):
+        # TODO bad layer access
+        chunk_slice, scale, scale_factors, chunk, is_last_chunk = coord
+
+        layer_name = get_layer_name_for_scale(scale)
+        layer = viewer.layers[layer_name]
+
+        # TODO this relies on the coincidence that node indices are 2 or 3 for
+        #      the image and volume members of an Image layer
+        node = viewer.window.qt_viewer.layer_to_visual[
```

Comment:
The simplest way to prevent spamming the `qt_viewer` warning, which happens to me in the `multiscale_mandelbulb` example.

```suggestion
        node = viewer.window._qt_viewer.layer_to_visual[
```